pylops/utils/signalprocessing.py

```diff
@@ -1,20 +1,158 @@
+        b3 = (4 - s) * (2 + s) * (3 + s) * (4 + s) / 420.0
+        b4 = (1 + s) * (2 + s) * (3 + s) * (4 + s) / 1680.0
+        return b0, b1, b2, b3, b4
+
+    def b5d(sig: float):
+        s = sig
+        b0 = -(
+            (2 - s) * (3 - s) * (4 - s)
+            + (1 - s) * (3 - s) * (4 - s)
+            + (1 - s) * (2 - s) * (4 - s)
+            + (1 - s) * (2 - s) * (3 - s)
+        ) / 1680.0
+        b1 = (
+            -((2 - s) * (3 - s) * (4 + s) + (4 - s) * (3 - s) * (4 + s) + (4 - s) * (2 - s) * (4 + s))
+            / 420.0
+            + (4 - s) * (2 - s) * (3 - s) / 420.0
+        )
+        b2 = (
+            -((3 - s) * (3 + s) * (4 + s) + (4 - s) * (3 + s) * (4 + s)) / 280.0
+            + (4 - s) * (3 - s) * (4 + s) / 280.0
+            + (4 - s) * (3 - s) * (3 + s) / 280.0
+        )
+        b3 = (
+            -((2 + s) * (3 + s) * (4 + s)) / 420.0
+            + (4 - s) * (3 + s) * (4 + s) / 420.0
+            + (4 - s) * (2 + s) * (4 + s) / 420.0
+            + (4 - s) * (2 + s) * (3 + s) / 420.0
+        )
+        b4 = (
+            (2 + s) * (3 + s) * (4 + s)
+            + (1 + s) * (3 + s) * (4 + s)
+            + (1 + s) * (2 + s) * (4 + s)
+            + (1 + s) * (2 + s) * (3 + s)
+        ) / 1680.0
+        return b0, b1, b2, b3, b4
+
+    for i1 in range(nw, n1 - nw):
+        for i2 in range(0, n2 - 1):
+            s = dip[i1, i2]
+            if order == 1:
+                b0d, b1d, b2d = b3d(s)
+                b0, b1, b2 = b3(s)
+                v = din[i1 - 1, i2 + 1] - din[i1 + 1, i2]
+                u1[i1, i2] += v * b0d
+                u2[i1, i2] += v * b0
+                v = din[i1 + 0, i2 + 1] - din[i1 + 0, i2]
+                u1[i1, i2] += v * b1d
+                u2[i1, i2] += v * b1
+                v = din[i1 + 1, i2 + 1] - din[i1 - 1, i2]
+                u1[i1, i2] += v * b2d
+                u2[i1, i2] += v * b2
+            else:
+                c0d, c1d, c2d, c3d, c4d = b5d(s)
+                c0, c1, c2, c3, c4 = b5(s)
+                v = din[i1 - 2, i2 + 1] - din[i1 + 2, i2]
+                u1[i1, i2] += v * c0d
+                u2[i1, i2] += v * c0
+                v = din[i1 - 1, i2 + 1] - din[i1 + 1, i2]
+                u1[i1, i2] += v * c1d
+                u2[i1, i2] += v * c1
+                v = din[i1 + 0, i2 + 1] - din[i1 + 0, i2]
+                u1[i1, i2] += v * c2d
+                u2[i1, i2] += v * c2
+                v = din[i1 + 1, i2 + 1] - din[i1 - 1, i2]
+                u1[i1, i2] += v * c3d
+                u2[i1, i2] += v * c3
+                v = din[i1 + 2, i2 + 1] - din[i1 - 2, i2]
+                u1[i1, i2] += v * c4d
+                u2[i1, i2] += v * c4
+
+
+def _conv_allpass(
+    din: np.ndarray,
+    dip: np.ndarray,
+    order: int,
+    u1: np.ndarray,
+    u2: np.ndarray,
+) -> None:
+    """Dispatch to numba kernel when available, otherwise use Python fallback."""
+    global _pwd_warning_emitted
```

Comment:
Can you explain the rationale of this? Would just 

```
if _jit_message_pwd is not None:
         warnings.warn(_jit_message_pwd)
        _conv_allpass_python(din, dip, order, u1, u2)
```

not work equally well?